Bottom: a6cd20ca60bd4114fbfa2d60fd23b2a0cbaed51f
Top:    5ad5952c4ae4e151b00cbdcfd8057dc9f279b110
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-04 16:51:03 -0500

Refresh of buildbot-worker-ptxdist

---
diff --git a/roles/buildbot-worker-ptxdist/defaults/main.yml b/roles/buildbot-worker-ptxdist/defaults/main.yml
index 9cdd8b6f..092944c9 100644
--- a/roles/buildbot-worker-ptxdist/defaults/main.yml
+++ b/roles/buildbot-worker-ptxdist/defaults/main.yml
@@ -2,22 +2,31 @@
 # defaults file for buildbot-worker-ptxdist
 
 # The hostname passed as an envirnment variable into the container
-container_addr_map:
-  buildbot:
-    ip_addr: '127.0.0.1'
-    hostname: buildbot-worker-ptxdist.example.com
-
 # Network names
 docker_network_frontend: 'frontend'
 
-buildbot_worker_ptxdist_data_path: '/buildbot'
-
-# the name of the image being duilt and used for the container
-buildbot_worker_ptxdist_image_repo: https://github.com/novatechweb/docker-buildbot-worker-ptxdist.git
-buildbot_worker_ptxdist_image_name: '{{ docker_registry_username }}/buildbot-worker-ptxdist'
+buildbot_worker_ptxdist_build_args:
+  BUILDBOT_UID: '1000'
+  BUILDBOT_VERSION: 'v1.1.2'
 
 # the name of the container being started
 buildbot_worker_ptxdist_container_name: '{{ docker_name_prefix }}buildbot-worker-ptxdist'
 
-# the name of the data-volume used by the container
-buildbot_worker_ptxdist_dv_name: '{{ docker_name_prefix }}buildbot-worker-ptxdist_DV'
+buildbot_worker_ptxdist_data_path: '/buildbot'
+
+# the name of the data volume used by the container
+buildbot_worker_ptxdist_data_volume: '{{ docker_name_prefix }}buildbot-worker-ptxdist_data'
+
+buildbot_worker_ptxdist_env:
+  BUILDBOT_MASTER: '{{ buildbot_master }}'
+  BUILDBOT_WORKER_PORT: '9989'
+  BUILDBOT_WORKER_NAME: 'worker-ptxdist'
+  BUILDBOT_WORKER_PASS: 'pass'
+
+# the name of the image being built and used for the container
+buildbot_worker_ptxdist_image_dir: '{{ docker_projects_dir }}/buildbot-worker-ptxdist'
+buildbot_worker_ptxdist_image_name: '{{ docker_registry_username }}/buildbot-worker-ptxdist'
+buildbot_worker_ptxdist_image_repo: https://github.com/novatechweb/docker-buildbot-worker-ptxdist.git
+
+buildbot_worker_ptxdist_volumes:
+  - '{{ buildbot_worker_ptxdist_data_volume }}:{{ buildbot_worker_ptxdist_data_path }}:z'
diff --git a/roles/buildbot-worker-ptxdist/tasks/main.yml b/roles/buildbot-worker-ptxdist/tasks/main.yml
index 9bc8c930..c2b6144a 100644
--- a/roles/buildbot-worker-ptxdist/tasks/main.yml
+++ b/roles/buildbot-worker-ptxdist/tasks/main.yml
@@ -8,23 +8,38 @@
 
 - name: Create buildbot data volume
   docker_volume:
-    name: '{{ buildbot_dv_name }}'
+    name: '{{ buildbot_worker_ptxdist_data_volume }}'
 
 - name: Checkout image repo
   git:
     repo: '{{ buildbot_worker_ptxdist_image_repo }}'
     version: master
-    dest: '{{ docker_projects_dir }}/buildbot-worker-ptxdist'
+    dest: '{{ buildbot_worker_ptxdist_image_dir }}'
 
-- name: Create buildbot master image
+- name: Create buildbot worker image
   docker_image:
     name: '{{ buildbot_worker_ptxdist_image_name }}'
     tag: '{{ docker_image_tag }}'
-    path: '{{ docker_projects_dir }}/buildbot-worker-ptxdist'
+    path: '{{ buildbot_worker_ptxdist_image_dir }}'
+    buildargs: '{{ buildbot_worker_ptxdist_build_args }}'
+    force: true
 
 - name: Start buildbot container
   docker_container:
     name: '{{ buildbot_worker_ptxdist_container_name }}'
     image: '{{ buildbot_worker_ptxdist_image_name }}:{{ docker_image_tag }}'
-    volumes:
-      - '{{ buildbot_worker_ptxdist_dv_name }}:{{ buildbot_worker_ptxdist_data_path }}:z'
+    # networks: '{{ buildbot_worker_ptxdist_networks }}'
+    volumes: '{{ buildbot_worker_ptxdist_volumes }}'
+    env: '{{ buildbot_worker_ptxdist_env }}'
+    # ports: '{{ buildbot_worker_ptxdist_port_args }}'
+
+- name: Grab ssh public key
+  command: >
+    docker exec '{{ buildbot_worker_ptxdist_container_name }}' cat /home/buildbot/.ssh/id_rsa.pub
+  changed_when: false
+  register: output
+
+- name: Add buildbot_worker_ptxdist host
+  add_host:
+    name: buildbot_worker_ptxdist
+    pubkey: "{{ output.stdout }}"
